### DISCUSSION TITLE
Running QZ on iPad with Apple Watch 10 and NordicTrack treadmill - which configs?

### DISCUSSION THREAD

**Discussion by magicdude4eva** [cagnulein/qdomyos-zwift#1946] [Q&A] [answered]
As the title says, my idea is that I run on the iPad Zwift or Apple Fitness+. I also have an AW10 and a NordicTrack treadmill. Can someone explain what setting-change I need to apply so that:
* Zwift / Apple Fitness sees the metrics from the treadmill
* Zwift / Apple Fitness sees the metrics (BPM etc) from the Apple Watch

For me it is unclear if I need to install the QZ companion app on the Watch or if it is enough to just have QZ to run on Apple iPad.

Any help would be appreciated.

**Comment by cagnulein** [comment]
in order to do this you need also to run qz on the iphone since the iphone will do the bridge to AW to the ipad. QZ on iphone must be just running, you don't need to configure it to connect to your treadmill, it has just to stay alive

**Reply by magicdude4eva** [reply]
Is there a way to simplify this? For example run QZ headless on a raspberry and then not need QZ to run on iPad and iPhone?

**Reply by cagnulein** [reply]
if you want to use the qz on apple watch, an iphone must be involved. so i suggest to use qz on iphone in your scenario.

so qz on the iphone and zwift on ipad or apple fitness on ipad

**Comment by magicdude4eva** [comment]
> if you want to use the qz on apple watch, an iphone must be involved. so i suggest to use qz on iphone in your scenario.
> 
> so qz on the iphone and zwift on ipad or apple fitness on ipad

Just to clarify:
* On the iPad I only run Zwift or Apple Fitness. QZ is not installed
* On the iPhone I have QZ running and paired to my treadmill. Do I need to make any other adjustments to settings if I do a clean install of QZ on the iPhone?
* On the Apple Watch I do not have QZ installed and just use the regular fitness apps (such as "Couch to 5K")

I guess that iPad would need to be Bluetooth-paired with the iPhone so that Zwift/Apple Fitness can pick up the data from the treadmill/Apple Watch?

Would I be able to achieve this also with a Raspberry Pi? Ideally I want my wife to just walk into the gym room, start the treadmill, do a workout on the Apple Watch and everything just works? I am looking to improve "WAF" (wife-approval-factor :-))

**Comment by cagnulein** [comment]
> * On the iPhone I have QZ running and paired to my treadmill. Do I need to make any other adjustments to settings if I do a clean install of QZ on the iPhone?

if you want to have heart rate from apple watch, you have to install the qz app on the watch too and start it too from there.



> Would I be able to achieve this also with a Raspberry Pi? Ideally I want my wife to just walk into the gym room, start the treadmill, do a workout on the Apple Watch and everything just works? I am looking to improve "WAF" (wife-approval-factor :-))

this can work using this branch https://github.com/cagnulein/qdomyos-zwift/pull/1915 check this thread https://github.com/cagnulein/qdomyos-zwift/issues/1913

**Comment by magicdude4eva** [comment]
If I run Zwift on iPad and then QZ on iPhone and QZ on Apple Watch I get all the devices to properly show up and this works great. Is there any way that the Zwift app can control the treadmill (I am running your Nordictrack S25 Testflight build).

**Reply by cagnulein** [reply]
yes , set the zwift credentials in qz and you will be ready to go!

**Reply by magicdude4eva** [reply]
In the meantime I have compiled it on the Pi - would those parameters be supported in "/root/.config/Roberto Viola/qDomyos-Zwift.conf" and if so which?

**Reply by cagnulein** [reply]
nope, zwift api aren't working on linux. only android or ios

**Reply by magicdude4eva** [reply]
Perfect - I have the Testflight version 2.16 (698), configured the Zwift credentials on my iPhone and started a workout. Should I then be able to control within Zwift the treadmill? I have captured debug logs and am more than happy to create an issue instead.
[debug-Thu_Jan_4_19_45_59_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/13834369/debug-Thu_Jan_4_19_45_59_2024.log)
[debug-Thu_Jan_4_19_41_49_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/13834370/debug-Thu_Jan_4_19_41_49_2024.log)

**Comment by cagnulein** [comment]
in the log i saw it worked correctly, it's simple in a 0% grade run. did
you do a free run with inclination? could you please do another one with
>0%?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 4 gen 2024 alle 19:54 Gerd Naschenweng <
***@***.***> ha scritto:

**Reply by magicdude4eva** [reply]
I tried a few different workouts and attached logs. The increase/decrease intensity does seemingly nothing. I did notice that the treadmill adjusts incline on some workouts. 
[debug-Thu_Jan_4_20_22_28_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/13834593/debug-Thu_Jan_4_20_22_28_2024.log)
[debug-Thu_Jan_4_20_16_38_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/13834598/debug-Thu_Jan_4_20_16_38_2024.log)
[debug-Thu_Jan_4_20_13_32_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/13834605/debug-Thu_Jan_4_20_13_32_2024.log)
[debug-Thu_Jan_4_20_08_14_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/13834609/debug-Thu_Jan_4_20_08_14_2024.log)
[debug-Thu_Jan_4_19_45_59_2024 2.log](https://github.com/cagnulein/qdomyos-zwift/files/13834611/debug-Thu_Jan_4_19_45_59_2024.2.log)

**Comment by cagnulein** [comment]
are you saying workout, but we are talking about free run right? it doesn't
work on workouts , only free runs
i will check your debug log tomorrow

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 4 gen 2024 alle 20:26 Gerd Naschenweng <
***@***.***> ha scritto:

**Reply by cagnulein** [reply]
in addition to my previous question @magicdude4eva I checked the log and in all of them the inclination it changes accordingly to zwift to your treadmill. could you please explain better your issue? thanks!

**Reply by magicdude4eva** [reply]
hi - I was expecting that the speed increases when I click on the "intensity" button - this changes the speed on Zwift, but not on the treadmill.

to me it is not clear what the Zwift integration does / does not do. I did notice that the inclination adjusted when I did a run this morning - there was however (at least to me) no visible indication in Zwift that the inclination changes. I thought that the run-speed would also adjust on the treadmill if I adjust it on Zwift.

so you are saying that workouts are not supported. is there a reason for it - perhaps I trivialise it but I thought that through the Zwift integration QZ gets webhook / polls current state and would then based on Zwift adjust the speed.

**Reply by cagnulein** [reply]
> hi - I was expecting that the speed increases when I click on the "intensity" button - this changes the speed on Zwift, but not on the treadmill.

what is the intensity button? in zwift you mean? so you were doing a workout? if so this is not involved in the zwift api that I added in the latest beta. For the workout you have to follow this https://github.com/cagnulein/qdomyos-zwift/wiki/Can-Zwift-control-speed-and-or-inclination-of-my-treadmill%3F

> to me it is not clear what the Zwift integration does / does not do. I did notice that the inclination adjusted when I did a run this morning - there was however (at least to me) no visible indication in Zwift that the inclination changes. I thought that the run-speed would also adjust on the treadmill if I adjust it on Zwift.

nope the integration just change the inclination following the grade showed in the right corner of zwift during a free ride. for workouts instead, you have to follow the above link with the PC version

> so you are saying that workouts are not supported. is there a reason for it - perhaps I trivialise it but I thought that through the Zwift integration QZ gets webhook / polls current state and would then based on Zwift adjust the speed.

yep because zwift doesn't send via API the metrics that you're doing in the workout. I'm literally hacking Zwift, there is no open standard or similar. So for now we have OCR on the PC side and ZWIFT api on mobile.

I hope it will be clear

**Reply by magicdude4eva** [reply]
Thank you for the explanation. I think the easiest setup right now for me is:
- Nordictrack S25 treadmill
- QZ running on iPhone
- QZ running on Apple Watch
- Zwift running on iPad Pro (and connection to iPhone)

Does the QZ companion app on the watch run and broadcast in the background? To me it seems that if I start the QZ companion app and then for example "Couch to 5K" on the watch, the QZ app goes into the foregrund - hence causing issues with the workout apps. Is there perhaps a specific sequence in powering up everything involved?

**Reply by cagnulein** [reply, marked as answer]
@magicdude4eva you can't run 2 apps for workouts on the watch at the same time, it's an apple limitation
